databases/week3/social_media.sql

```
@@ -0,0 +1,106 @@
+('Bob Smith', 'securepass', 'bob@example.com', '2023-02-01'),
+('Charlie Brown', 'mypassword', 'charlie@example.com', '2023-03-01'),
+('Diana Prince', 'wonderwoman', 'diana@example.com', '2023-04-01'),
+('Eve Adams', 'evepass', 'eve@example.com', '2023-05-01');
+
+INSERT INTO post (user2_id, content, created_date, updated_date, relation_to_author) VALUES
+(1, 'This is Alices first post!', '2023-01-02', NULL, 'personal'),
+(2, 'Bob shares his thoughts on technology.', '2023-02-05', NULL, 'professional'),
+(3, 'Charlie posts about his favorite books.', '2023-03-10', '2023-03-15', 'hobby'),
+(4, 'Diana shares her travel experiences.', '2023-04-12', NULL, 'travel'),
+(5, 'Eve writes about her cooking adventures.', '2023-05-20', NULL, 'food');
+
+INSERT INTO comment (post_id, user2_id, content, created_date, updated_date, relation_to_author, relation_to_post, relation_to_optional_parent_comment) VALUES
+(1, 2, 'Great post, Alice!', '2023-01-03', NULL, 'friend', 'positive', NULL),
+(2, 3, 'Interesting perspective, Bob.', '2023-02-06', NULL, 'colleague', 'neutral', NULL),
+(3, 4, 'I love this book too, Charlie!', '2023-03-11', NULL, 'friend', 'positive', NULL),
+(4, 5, 'Amazing travel story, Diana!', '2023-04-13', NULL, 'follower', 'positive', NULL),
+(5, 1, 'Can\'t wait to try this recipe, Eve!', '2023-05-21', NULL, 'friend', 'positive', NULL);
+
+INSERT INTO reaction (post_id, user2_id, comment_id, type, created_date, updated_date, relation_to_author) VALUES
+(1, 2, 1, 'like', '2023-01-03', NULL, 'friend'),
+(2, 3, NULL, 'highfive', '2023-02-06', NULL, 'colleague'),
+(3, 4, 3, 'laugh', '2023-03-11', NULL, 'friend'),
+(4, 5, NULL, 'like', '2023-04-13', NULL, 'follower'),
+(5, 1, 5, 'cry', '2023-05-21', NULL, 'friend');
+
+INSERT INTO message (sender_id, receiver_id, content, created_date, updated_date, relation_to_author, connection_id) VALUES
+(1, 2, 'Hi Bob, how are you?', '2023-01-02', NULL, 'friend', 1),
```

The ⁠INSERT INTO message statements reference `⁠connection_id` values (1 through 5). However, there's no mock data for the ⁠connection table. This means these inserts would fail due to foreign key violations if the ⁠connection table is empty.